### PR TITLE
Adds functionality to the Regeneration superpower

### LINKED ISSUE #16966: M_REGEN ("You feel better" genetics mutation) does literally nothing.
As in there's not even any code for it.

Should either be made to do something or commented out

### PR DESCRIPTION
**Pull request by boy2mantwicethefam**
[tested]
As it turns out M_REGEN quite literally did nothing, it was a placebo superpower, and the one comment I managed to find about it mentioned it's supposed to double the regeneration but makes you go hungry, so there's that.
Regeneration superpower will now double the amount of damage you regenerate but will make you go hungry twice as fast. Also applies for blood regeneration. It can PROBABLY out-regenerate punches if the user is fed but it cannot out-regenerate anything higher.
I'm not really sure if this actually heals twice the amount or far more considering that heal_amt is a variable actually I'll look into it in a bit but you get the idea of what the PR does I know that it multiplies that whole heal_amt from my testing being an overfed person healed like 1,5 brute per tick once the wound was treated actually I'm not sure if this whole thing works weirdly enough I haven't looked at the external organ (limb) code something something I'm fixing it
fixes #16966
:cl:
 * rscadd: The Regeneration superpower now has a functionality! Previously, it did nothing, but now it will make you regenerate twice as fast but make you go hungry twice as fast!

### CODE REVIEW

File: code/modules/organs/organ_external.dm [hunk 1]
```diff
@@ -596,7 +596,7 @@ Note that amputating the affected organ does in fact remove the infection from t
 		//Configurable regen speed woo, no-regen hardcore or instaheal hugbox, choose your destiny
 		heal_amt = heal_amt * config.organ_regeneration_multiplier
 		if(M_REGEN in owner.mutations)
-			heal_amt = heal_amt * config.organ_regeneration_multiplier //Regenerates a second time if you have regeneration
+			heal_amt = heal_amt * 2 // Will make you regenerate twice as fast if you have regeneration
```
Comment (jknpj) [inline code/modules/organs/organ_external.dm]: Don't change a var to a magic number, make config.organ_regeneration_multiplier 2 instead.

**Review by jknpj — DISMISSED** [review]

File: code/modules/organs/organ_external.dm (hunk 1 above)
Comment (boy2mantwicethefam) [reply code/modules/organs/organ_external.dm]: Sure, but from what I see config.organ_regeneration is just one thing that adds up to heal_amt

Comment (boy2mantwicethefam) [reply code/modules/organs/organ_external.dm]: Actually now that I think about it you're right, from what I see though it's the exact same thing

File: code/modules/organs/organ_external.dm [hunk 2]
```diff
@@ -596,7 +596,7 @@ Note that amputating the affected organ does in fact remove the infection from t
 		//Configurable regen speed woo, no-regen hardcore or instaheal hugbox, choose your destiny
 		heal_amt = heal_amt * config.organ_regeneration_multiplier
 		if(M_REGEN in owner.mutations)
-			heal_amt = heal_amt * 2 // Will make you regenerate twice as fast if you have regeneration
+			heal_amt = heal_amt * config.organ_regeneration_multiplier // Will heal based on organ_regeneration_multiplier twice if you have Regeneration superpower
```
Comment (Exxion) [inline code/modules/organs/organ_external.dm]: So it *squares* the healing rate? With default config that's the same as doubling it, but relying on that isn't great

Comment (boy2mantwicethefam) [reply code/modules/organs/organ_external.dm]: I could cook up a new config called regeneration multiplier and set that to multiply by 2

Comment (Exxion) [reply code/modules/organs/organ_external.dm]: You could but you could also just multiply it by two since that's what you intend to do anyway

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
Fixes #16966

**Comment by MadmanMartian** [comment]
probably best if #21512 is merged first, so it can use the variable there to speed up hunger

**Pull request merged**